awx/ui/client/src/templates/workflows/workflow-maker/workflow-maker.controller.js

```diff
@@ -155,49 +153,49 @@ export default ['$scope', 'WorkflowService', 'GetBasePath', 'TemplatesService',
+                        data: buildSendableNodeData()
+                    })
+                    .then(function (data) {
 
-                    if (!params.node.isRoot) {
-                        associateRequests.push({
-                            parentId: params.parentId,
-                            nodeId: data.data.id,
-                            edge: params.node.edgeType
-                        });
-                    }
+                        if (!params.node.isRoot) {
+                            associateRequests.push({
+                                parentId: params.parentId,
+                                nodeId: data.data.id,
+                                edge: params.node.edgeType
+                            });
+                        }
 
-                    if (_.get(params, 'node.promptData.launchConf.ask_credential_on_launch')){
-                         // This finds the credentials that were selected in the prompt but don't occur
-                         // in the template defaults
-                         let credentialsToPost = params.node.promptData.prompts.credentials.value.filter(function(credFromPrompt) {
-                             let defaultCreds = params.node.promptData.launchConf.defaults.credentials ? params.node.promptData.launchConf.defaults.credentials : [];
-                             return !defaultCreds.some(function(defaultCred) {
-                                 return credFromPrompt.id === defaultCred.id;
-                             });
-                         });
-
-                         credentialsToPost.forEach((credentialToPost) => {
-                             credentialRequests.push({
-                                 id: data.data.id,
-                                 data: {
-                                     id: credentialToPost.id
-                                 }
-                             });
-                         });
-                     }
-
-                    params.node.isNew = false;
-                    continueRecursing(data.data.id);
-                }, function(error) {
-                    ProcessErrors($scope, error.data, error.status, null, {
-                        hdr: 'Error!',
-                        msg: 'Failed to add workflow node. ' +
-                        'POST returned status: ' +
-                        error.status
+                        if (_.get(params, 'node.promptData.launchConf.ask_credential_on_launch')) {
+                            // This finds the credentials that were selected in the prompt but don't occur
+                            // in the template defaults
+                            let credentialsToPost = params.node.promptData.prompts.credentials.value.filter(function (credFromPrompt) {
+                                let defaultCreds = params.node.promptData.launchConf.defaults.credentials ? params.node.promptData.launchConf.defaults.credentials : [];
```

Comment:
can also do `_.get(params, 'node.promptData.launchConf.defaults.credentials', [])` for this kind of thing.